examples/pipelines/hf_dataset_pipeline/components/load_from_hub/src/main.py

```diff
@@ -0,0 +1,49 @@
+"""
+This component uploads a Hugging Face dataset from the hub to a Google Cloud Storage bucket.
+"""
+from typing import Optional, Union, Dict
+
+import pandas as pd
+from datasets import Dataset, load_dataset, concatenate_datasets
+
+from express.components.hf_datasets_components import HFDatasetsLoaderComponent, HFDatasetsDatasetDraft
+from express.logger import configure_logging
+
+
+class SeedDatasetLoader(HFDatasetsLoaderComponent):
+    """Class that inherits from Hugging Face data loading """
+
+    @classmethod
+    def load(cls, extra_args: Optional[
+        Dict[str, Union[str, int, float, bool]]] = None) -> HFDatasetsDatasetDraft:
+        """
+        An example function showcasing the data loader component using Express functionalities
+        Args:
+            extra_args (Optional[Dict[str, Union[str, int, float, bool]]): optional args to pass to
+             the function (e.g. seed data source)
+        Returns:
+            HFDatasetsDatasetDraft: a dataset draft that creates a plan for an output datasets/manifest
+        """
+        configure_logging()
+        
+        # 1) Create data source
+        caption_dataset = load_dataset(extra_args["dataset_name"], split="train")
+
+        # 2) Create an example index
+        index_list = [f"image_{idx}" for idx in range(len(caption_dataset))]
+        index_df = pd.DataFrame(index_list, columns=['index'])
+        index = Dataset.from_pandas(index_df)
+
+        caption_dataset = concatenate_datasets([index, caption_dataset])
+        
+        # 2.2) Create data_source dictionary
+        data_sources = {"captions": caption_dataset}
```

Comment:
ok for the initial loading but we might want to think of a more efficient way of doing this. Right now we're materializing the dataset in both the `data_sources` dict and also the `caption_dataset` which can lead to OoO issues with large datasets

Reply:
I don't think this is the case since both are referring to the same object.

But the current setup is indeed limited by the local memory, so we should find a way to stream this data instead.

Reply:
Does Python duplicate the memory when using a variable in a dict?

Reply:
@RobbeSneyders you're right I think it just references it. I would still opt to use streaming instead 
https://huggingface.co/docs/datasets/stream

Reply:
Streaming is mostly useful when you'd like to transform the dataset, not when you want to upload it to the cloud like here, so will keep it like it is. However for our transform components we can leverage streaming